src/control/server/mgmt_client.go

```diff
@@ -176,7 +175,7 @@ func (msc *mgmtSvcClient) PrepShutdown(ctx context.Context, destAddr string, req
 			resp, err = pbClient.PrepShutdownRanks(ctx, &req)
 
 			return
-		})
+		}, grpc.FailOnNonTempDialError(true))
```

Comment:
Wondering, should this be part of a shared helper function? Should we do this for every gRPC call?

Reply:
now we don't want to do it for join, could create helper functions for e.g. WithConnectionRedial and just WithConnection